src/main/java/com/cyber/mysticism/tarot/service/ThreeCardsDivinationService.java

```diff
@@ -0,0 +1,87 @@
+package com.cyber.mysticism.tarot.service;
+
+import com.cyber.mysticism.tarot.json.Card;
+import com.cyber.mysticism.tarot.repository.TarotDeckRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+public class ThreeCardsDivinationService {
+
+    public static final String THREE_CARDS_DIVINATION = "three_cards_divination";
+    public static final String MAJOR_ARCANA = "Major Arcana";
+    private static final Logger logger = LoggerFactory.getLogger(ThreeCardsDivinationService.class);
+    private final TarotDeckRepository tarotDeckRepository;
+
+    @Autowired
+    public ThreeCardsDivinationService() throws DivinationException {
+        this.tarotDeckRepository = TarotDeckRepository.getInstance();
+    }
+
+    private static Collector<Card, Object, Stream<Card>> toShuffledStream() {
+        return Collectors.collectingAndThen(Collectors.toList(), collected -> {
+            Collections.shuffle(collected);
+            return collected.stream();
+        });
```

Comment:
**Suggestion:** Consider using `java.util.concurrent.ThreadLocalRandom` for shuffling the cards instead of `Collections.shuffle()`. `ThreadLocalRandom` is more efficient in a multi-threaded environment. [performance]
```suggestion
    private static Collector<Card, Object, Stream<Card>> toShuffledStream() {
        return Collectors.collectingAndThen(Collectors.toList(), collected -> {
            ThreadLocalRandom.current().ints(0, collected.size()).distinct().limit(collected.size()).forEach(i -> Collections.swap(collected, i, ThreadLocalRandom.current().nextInt(collected.size())));
            return collected.stream();
        });
    }
```